symbolic-symcache/tests/test_transformers.rs

```diff
@@ -0,0 +1,91 @@
+use std::fmt;
+use std::io::Cursor;
+
+use symbolic_common::{ByteView, SelfCell};
+use symbolic_debuginfo::macho::BcSymbolMap;
+use symbolic_debuginfo::Object;
+use symbolic_symcache::transform::{self, Transformer};
+use symbolic_symcache::{SymCache, SymCacheWriter};
+
+type Error = Box<dyn std::error::Error>;
+
+/// Helper to create neat snapshots for symbol tables.
+struct FunctionsDebug<'a>(&'a SymCache<'a>);
+
+#[allow(deprecated)]
+impl fmt::Debug for FunctionsDebug<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut vec: Vec<_> = self
+            .0
+            .functions()
+            .filter_map(|f| match f {
+                Ok(f) => {
+                    if f.address() != u32::MAX as u64 {
+                        Some(f)
+                    } else {
+                        None
+                    }
+                }
+                Err(_) => None,
+            })
+            .collect();
+
+        vec.sort_by_key(|f| f.address());
+        for function in vec {
+            writeln!(f, "{:>16x} {}", &function.address(), &function.name())?;
+        }
+
+        Ok(())
+    }
+}
+
+// FIXME: This is a huge pain, can't this be simpler somehow?
+struct OwnedBcSymbolMap(SelfCell<ByteView<'static>, BcSymbolMap<'static>>);
+
+impl Transformer for OwnedBcSymbolMap {
+    fn transform_function<'f>(&'f self, f: transform::Function<'f>) -> transform::Function<'f> {
+        self.0.get().transform_function(f)
+    }
+
+    fn transform_source_location<'f>(
+        &'f self,
+        sl: transform::SourceLocation<'f>,
+    ) -> transform::SourceLocation<'f> {
+        self.0.get().transform_source_location(sl)
+    }
+}
+
+#[test]
+fn test_transformer_symbolmap() -> Result<(), Error> {
+    let buffer = ByteView::open(
+        "../symbolic-debuginfo/tests/fixtures/2d10c42f-591d-3265-b147-78ba0868073f.dwarf-hidden",
```

Comment:
in theory this could be moved to fixtures, but it would have to be copied. it looks like doctests reference these files, so they can't just be deleted. i'm thinking it's better just to keep this as-is to avoid any drift between the doctest data and the unit test data.